### DISCUSSION TITLE
Which variant uno project to use for a multi-platform app?

### DISCUSSION THREAD

**Discussion by Jyakomo** [unoplatform/uno#15396] [Q&A]
New to uno platform. Having previously used Visual Studio for a windows .net app with the toolbox for controls; however, to create a multi-platform equivalent app, which uno variant is recommended? There are 4 variants on the tutorial: XAML+MVUX, XAML+MVVM, C# Markup+MVUX, C# Markup+MVVM.  To utilize the C# code from my windows .net app, which variant s recommended? And does the uno platform support a toolbox (for controls)?  Thank you for all your help

**Comment by francoistanguay** [comment]
You first need to choose between XAML and C# for your UI language. If you're familiar with XAML, you might want to start there. However, we now shipped C# Markup that offers the exact same API as the XAML-based API (Object Model) but entirely in C#, using a Fluent API and Source Generators.  It allows you to use C# for your UI as well as the rest of your application logic and skip XAML entirely.

C# Markup docs: https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Markup/Overview.html

Once you chose XAML or C# for your UI language, you can select your preferred state management approach. Most people become first familiar with MVVM which allows for great decoupling from the View.

We now introduced a new pattern called MVUX that has all the benefits of MVVM without all the hassle of dealing with ICommand, INotifyPropertyChanged, ... and also takes care of Threading considerations, making sure everything is async, in an immutable-friendly world. 

MVUX Docs: https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/Overview.html?tabs=viewmodel%2Cmodel

In all cases, I recommend you try our Counter and SimpleCalc samples here: https://platform.uno/docs/articles/getting-started/counterapp/get-started-counter.html

It will give you a great sense of resemblances and differences.

If you ask me personally, I would definitely try to learn our MVUX as we believe it's the cleanest way to do state management in an app in .NET, and C# Markup has gained quite a bit of popularity in our audience.

As for Toolbox, I'm not sure what you mean but if you mean a Toolbox of Components in the context of a Designer, there's is none at this point. Our community leverages and relies heavily on our great Hot Reload: https://platform.uno/docs/articles/features/working-with-xaml-hot-reload.html?tabs=vswin%2Cskia

Enjoy!
